### DISCUSSION TITLE
Contentlayer 2, community and roadmap

### DISCUSSION THREAD

**Discussion by timlrx** [timlrx/contentlayer2#3] [Announcements]
Hi Contentlayer community,

As you might already know [maintenance of Contentlayer has stalled](https://github.com/contentlayerdev/contentlayer/issues/429) since Netlify's acquistion. This is a new community managed updated fork that aims to keep updated the features that everyone loves including:

- Live reload on content changes
- Fast and incremental builds
- Simple but powerful schema DSL to design your content model (validates your content and generates types)
- Auto-generated TypeScript types based on your content model (e.g. frontmatter or CMS schema)
- Local content (Markdown, MDX, JSON, YAML)
- Next.js

The first task is to update all the broken dependencies and packages. This has been [completed](https://github.com/timlrx/contentlayer2/pull/1) and the new packages are published as contentlayer2. If you are an existing Contentlayer user using, try upgrading your packages to the following: 

- [@contentlayer2](https://www.npmjs.com/settings/contentlayer2/packages)
- [contentlayer2](https://www.npmjs.com/package/contentlayer2)
- [next-contentlayer2](https://www.npmjs.com/package/next-contentlayer2)

The next task is to update core documentation and examples which are down or that might be affected by the migration. In the meantime feel free to refer to https://contentlayer.aelliixx.com/ for a temporary docs. You might need to change the package reference to `contentlayer2` instead of `contentlayer` as in the docs.

Once that is done, we can turn out attention to making other code related changes e.g. updating the core effect code to the new stable version or many of the outstanding pull request or feature requests. This might also mean cutting down support for some of the existing content sources e.g. Stackbit, Contentful or Notion. 

I know that there are a few projects trying to create a better / new version of Contentlayer as well. Happy for those to remain as independent projects or for future ideas to be merged.

What if one day Netlify or Schickling decides to continue with Contentlayer? I will be happy to merge the code back to the original if the spirit of the original project is maintained. 

Contentlayer is a large project to support and maintain. Happy to get on board new project maintainers who are keen on contributing and maintaining. Let me know over here or via PM if you are interested. If there are any ideas or feedback with regards to community management or the roadmap, feel feel to post them over here.

Edit: It seems like there are plans to continue with the project: https://github.com/contentlayerdev/contentlayer/issues/651#issuecomment-2030335434. In that case, the fork will focus more on bug fixes and updates rather than any breaking changes or new features. In the meantime, enjoy the stay 🌴 

Thanks,
Tim

**Comment by gaurangrshah** [comment]
Appreciate you taking this on @timlrx I can help with docs and stuff and willing to lend a hand if needed as time allows.

**Reply by timlrx** [reply]
Thanks, as mentioned by Sean in https://github.com/contentlayerdev/contentlayer/issues/651#issuecomment-2030335434, it seems like they are looking to continue with the project. In that case, `contentlayer2` will just be focused on bug fixes while keeping to the same functionalities provided by the original i.e. we will continue to use the original docs.

Feel free to open a PR or suggest a fix if you encounter any problems along the way.

**Comment by Singh-Prince-Rinku** [comment]
Response #2

**Comment by PxlSyl** [comment]
Thank you @timlrx , you're amazing!!

**Comment by git-create-devben** [comment]
Thank you @timlrx you're a live saver

**Comment by TheMikeyRoss** [comment]
Thank you @timlrx , i think this discussion should be linked in the README so people know how to migrate

**Comment by myreceiptt** [comment]
#ReallyVeryBigThanks #OiOi!
Now I can try upgrading my packages to ContentLayer2.
Wish me luck!

**Comment by nixjs** [comment]
Thank you @timlrx . And how can i PR my website?

**Reply by timlrx** [reply]
Add it over here: https://github.com/timlrx/contentlayer2?tab=readme-ov-file#who-is-using-contentlayer
